core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -108,10 +129,19 @@ public static void writeSegmentFile(String tablePath, final String taskNo, Strin
         folderDetails.setPartitions(partionNames);
         folderDetails.setStatus(SegmentStatus.SUCCESS.getMessage());
         for (CarbonFile file : carbonFiles) {
-          folderDetails.getFiles().add(file.getName());
+          if (file.getName().endsWith(CarbonTablePath.MERGE_INDEX_FILE_EXT)) {
+            folderDetails.setMergeFileName(file.getName());
+          } else {
+            folderDetails.getFiles().add(file.getName());
+          }
         }
         segmentFile.addPath(location, folderDetails);
-        String path = writePath + "/" + taskNo + CarbonTablePath.SEGMENT_EXT;
+        String path = null;
+        if (isMergeIndexFlow) {
+          path = writePath + "/" + CarbonUtil.generateUUID() + CarbonTablePath.SEGMENT_EXT;
```

Comment:
Please write comments why you need a  UUID

Reply:
done